src/puppeteer_pool.js

```diff
@@ -120,6 +120,9 @@ export default class PuppeteerPool {
         this.activeInstances = {};
         this.retiredInstances = {};
         this.instanceKillerInterval = setInterval(() => this._killRetiredInstances(), instanceKillerIntervalMillis);
+
+        // ensure termination on SIGINT
+        this._terminateInstancesOnSigint();
```

Comment:
We should also remove `process.on` listener in `.destroy()` function. Something like:

```javascript
this.sigintListener = () => this._killAllInstances();
process.on('SIGINT', this.sigintListener);
```

and add 

```javascript
process.removeListener(this.sigintListener);
```

to `destroy()`.

We have usually destroy() function in the most of the projects which cleanup resources - intervals, timeouts, eventListeners, ... .